WeakAuras/TSUHelpers.lua

```diff
@@ -140,12 +140,41 @@ local createOrReplace = function(states, key, newState)
   end
 end
 
+local isChanged = function(states)
+  return states.__changed == true
+end
+
+local setChanged = function(states, changed)
+  states.__changed = changed
```

Comment:
If directly set "changedStates" here instead of in newIndex, and change the checking code to call IsChanged(), you don't need to do change "__index" and "__newIndex"

Reply:
Indeed, it's better